dot4gravity/src/lib.rs

```diff
@@ -223,16 +223,16 @@ pub struct GameState<Player> {
     /// Game mode.
     pub phase: GamePhase,
     /// When present,it contains the player that won.
-    pub winner: Option<PlayerIndex>,
+    pub winner: Option<Player>,
     /// Next player turn.
-    pub next_player: PlayerIndex,
+    pub next_player: Player,
     /// Players:
     pub players: [Player; NUM_OF_PLAYERS],
     /// Number of bombs available for each player.
     pub bombs: [(Player, u8); NUM_OF_PLAYERS],
```

Comment:
It think that have `winner`, `next_player` and `bombs` use `PlayerIndex` instead of player will make it more clear that those three fields depend on what's stored in the `players` field, it will also help in reducing the amount of clones we need to done, possibly even allow us to remove the `Clone` restrictions on some implementations and methods for `Player`

Reply:
I agree but the frontend client needs to display the information as they are without indexing. We're using more chain storage to persist these Player types but each game state gets removed at the end so should be alright.

Reply:
Actually let's ask if this is required, otherwise let's just close this PR :)

Reply:
Yeah, if winner contained `PlayerIndex` then we would, so then it's better to leave it as `Player` if that is a concern

Reply:
Cool let's stick to `Player` for clarity for now.